staging/src/k8s.io/apiserver/pkg/apis/audit/v1/types.go

```diff
@@ -31,6 +31,13 @@ const (
 	HeaderAuditID = "Audit-ID"
 )
 
+const (
+	// OmitManagedFieldsAnnotationKey is the name of the annotation for a Policy object, if
+	// set to 'true', the audit backend will omit 'managedFields' from the audit entries
+	// for both request and response.
+	OmitManagedFieldsAnnotationKey = "audit.kubernetes.io/omit-managed-fields"
```

Comment:
@lavalamp - I suggested moving this annotation key to the API, but I'm not 100% convinced.
But either way - I would appreciate your feedback (and if case it's needed review for it).

[I guess we want to add that in other versions too then]

Reply:
I agree this would be a lot better as an actual field on the policy object.

Reply:
@lavalamp please let me know if this looks good to you

```
type Policy struct {
	// +optional
	OmitManagedFields bool `json:"omitManagedFields,omitempty" protobuf:"varint,4,opt,name=omitManagedFields"`
}

type PolicyRule struct {
	// +optional
	OmitManagedFields *bool `json:"omitManagedFields,omitempty" protobuf:"varint,4,opt,name=omitManagedFields"`
}
```

it's in both `Policy` and `PolicyRule` so the cluster operator has an option to use it in a more granular way:
- the `bool` in `Policy` dictates the global default policy.
- the `*bool` in `PolicyRule` can override the global default for a particular matching rule, with three interpretations:
  - `nil` will represent 'not specified by the operator', hence use the global default.
  - `{false|true}` will override the global default.

Reply:
Yes, that sounds great, thanks!